workspaces/ui-v2/src/clients/spectacle.ts

```diff
@@ -47,6 +47,50 @@ export class SpectacleClient {
     return results.data.endpoint.commands.remove;
   };
 
+  public fetchFieldEditCommands = async (
+    fieldId: string,
+    requestedTypes: JsonType[]
+  ): Promise<CQRSCommand[]> => {
+    type FieldCommands = {
+      field: {
+        commands: {
+          edit: CQRSCommand[];
+        };
+      } | null;
+    };
+
+    const results = await this.spectacle.query<
+      FieldCommands,
+      {
+        fieldId: string;
+        requestedTypes: JsonType[];
+      }
+    >({
+      query: `
+      query X($fieldId: ID!, $requestedTypes: [JsonType!]!) {
+        field(fieldId: $fieldId) {
+          commands {
+            edit(requestedTypes: $requestedTypes)
+          }
+        }
+      }`,
+      variables: {
+        fieldId,
+        requestedTypes,
+      },
+    });
+    if (results.errors) {
+      console.error(results.errors);
+      throw new Error(JSON.stringify(results.errors));
+    }
+    if (!results.data || !results.data.field) {
+      const message = `Could not generate edit commands for field: ${fieldId}`;
+      console.error(message);
+      throw new Error(message);
```

Comment:
Just curious: why both `console.error` and throwing? Wouldn't doing just the latter also show up in console?

Reply:
It depends on how we error handle, redux catches these thrown errors and converts them into a rejected action. Uncaught errors (which crash the page) will show up in the console, we probably want to avoid this, but there's probably a better integration we can add in here